overlord/state/taskrunner.go

```diff
@@ -75,6 +79,22 @@ func NewTaskRunner(s *State) *TaskRunner {
 	}
 }
 
+// NewUnknownTaskRunner creates a new TaskRunner which handles all tasks *but* those listed in taskKinds.
+// This effectively intercepts all unknown tasks.
+func NewUnknownTaskRunner(s *State, taskKinds []string) *TaskRunner {
```

Comment:
Just in the few lines above in the review we already have three different names for the same idea: taskKinds, knownKinds, knownTasks. Can we please pick a single one and run with it?  Otherwise we need to keep in mind that all of these are actually the same thing across the code. As a suggestion, _knownTasks_ sounds good as it reflects the name of the manager as well.

Reply:
Note that this comment is perhaps irrelevant given the follow ups.